timemachine/cpp/src/kernels/k_local_md.cuh

```diff
@@ -26,3 +26,7 @@ void __global__ k_construct_bonded_params(
     params[idx * 3 + 1] = r_min;
     params[idx * 3 + 2] = r_max;
 }
+
+template <typename T> void __global__ k_update_indice(T *d_array, std::size_t idx, unsigned int val) {
+    d_array[idx] = val;
+}
```

Comment:
nit: surprise template -- is it intended to allow T other than unsigned int?

consider

```suggestion
void __global__ k_update_index(unsigned int *d_array, std::size_t idx, unsigned int val) {
    d_array[idx] = val;
}
```

or 

```suggestion
template <typename T> void __global__ k_update_index(T *d_array, std::size_t idx, T val) {
    d_array[idx] = val;
}
```

Reply:
Removed in https://github.com/proteneer/timemachine/pull/1014/commits/d8b0b8f5022d0f37036ca8b19d6519b8fd73f412